internal/provider/resource/project_resource.go

```diff
@@ -247,6 +269,14 @@ func (r *projectResource) Update(ctx context.Context, req resource.UpdateRequest
 		return
 	}
 
+	if state.ShouldCreateDefaultEnvs != plan.ShouldCreateDefaultEnvs {
+		resp.Diagnostics.AddError(
+			"Unable to update project",
+			"should_create_default_envs cannot be updated",
```

Comment:
to be more clear: should_create_default_envs cannot be modified after the resource is already created